xmake/modules/package/manager/pacman/find_package.lua

```diff
@@ -113,7 +113,9 @@ function main(name, opt)
 
     -- for msys2/mingw? mingw-w64-[i686|x86_64]-xxx
     if is_subhost("msys") and opt.plat == "mingw" then
-        name = (opt.arch == "x86_64" and "mingw-w64-x86_64-" or "mingw-w64-i686-") .. name
+        -- try to get the package prefix from the environment first
+        -- https://www.msys2.org/docs/package-naming/
+        name = (os.getenv("MINGW_PACKAGE_PREFIX") or (opt.arch == "x86_64" and "mingw-w64-x86_64" or "mingw-w64-i686")) .. "-" .. name
```

Comment:
why? it will break if I pass `add_requires("pacman::xxx", {arch = "i386"})` in mingw/x86_64 shell

Reply:
It is possible for a user to explicitly select non-current arch in add_requires.

Reply:
What if we checked opt.arch against os.arch instead?

Reply:
e.g.
```lua
os.arch() == opt.arch and os.getenv("MINGW_PACKAGE_PREFIX")
```

Reply:
As it is right now, every external package fails in msys2 unless you install the containing group which wastes space for environments that aren't in use. I'll keep thinking about this some more.

Reply:
I've found an alternative that might work. Please look over it :-)